scalding-core/src/main/java/com/twitter/scalding/cascading_interop/FlowListenerPromise.java

```diff
@@ -34,17 +36,24 @@
     final Promise<T> result = Promise$.MODULE$.<T>apply();
     flow.addListener(new FlowListener() {
       public void onStarting(Flow f) { } // ignore
-      public void onStopping(Flow f) { } // ignore
+      public void onStopping(Flow f) { // in case of runtime exception cascading call onStopping
+        if (!result.isCompleted()) {
+          result.failure(new Exception("Flow was stopped"));
```

Comment:
can we exercise this in a test where we throw in a FlowListener?

Reply:
done, thanks